src/Object/Image.php

```diff
@@ -775,25 +775,37 @@ public function getBlurHash(): string
 	 */
 	public function getFromBlurHash(string $blurhash, int $width, int $height)
 	{
-		if ($this->isImagick()) {
-			// Imagick is not supported
-			return;
-		}
-
 		$scaled = Images::getScalingDimensions($width, $height, 90);
 		$pixels = Blurhash::decode($blurhash, $scaled['width'], $scaled['height']);
 
-		$this->image = imagecreatetruecolor($scaled['width'], $scaled['height']);
+		if ($this->isImagick()) {
+			$this->image = new Imagick();
+			$draw  = new ImagickDraw();
+			$this->image->newImage($scaled['width'], $scaled['height'], '', 'png');
+		} else {
+			$this->image = imagecreatetruecolor($scaled['width'], $scaled['height']);
+		}
+
 		for ($y = 0; $y < $scaled['height']; ++$y) {
 			for ($x = 0; $x < $scaled['width']; ++$x) {
 				[$r, $g, $b] = $pixels[$y][$x];
-				imagesetpixel($this->image, $x, $y, imagecolorallocate($this->image, $r, $g, $b));
+				if ($this->isImagick()) {
+					$draw->setFillColor("rgb($r, $g, $b)");
+					$draw->point($x, $y);
+				} else {
+					imagesetpixel($this->image, $x, $y, imagecolorallocate($this->image, $r, $g, $b));
+				}
 			}
 		}
 
-		$this->width  = imagesx($this->image);
-		$this->height = imagesy($this->image);
-		$this->valid  = true;
+		if (!$this->isImagick()) {
+			$this->width  = imagesx($this->image);
+			$this->height = imagesy($this->image);
+		} else {
+			$this->image->drawImage($draw);
+		}
```

Comment:
Please avoid using a negative condition by swapping the statements.
```suggestion
		if ($this->isImagick()) {
			$this->image->drawImage($draw);
		} else {
			$this->width  = imagesx($this->image);
			$this->height = imagesy($this->image);			
		}
```

Reply:
Done